hack/build.sh

```diff
@@ -0,0 +1,89 @@
+# To push to your own registry, override the REGISTRY and NAMESPACE env vars,
```

Comment:
Could you please restructure the script to follow the convention used elsewhere in the project 
E.g. 
* https://github.com/rhobs/observability-operator/blob/main/test/run-e2e.sh
* https://github.com/rhobs/observability-operator/blob/main/hack/kind/setup.sh

```
#!/usr/bin/env bash 
set -eu -o pipefail

# traps 


# exports <if needed> 
# avoid exports if possible and make use of 
   FOOBAR=X \ 
   BAR=Baz \
   yq eval -i ...

# Globals 
declare FOO_BAR=${FOOBAR:-"default"}

# Constants 
declare -r READONLY_VAR="something"


# functions 
foo() {
}


main() {
}

main "$@"
```

Keeping all actions in seperate functions makes it easier to reason with